src/components/TokenLogo/index.js

```diff
@@ -48,28 +41,18 @@ const StyledHelpCircle = styled(HelpCircle)`
   background-color: ${({ theme }) => theme.white};
 `
 
-export function getCustomLogo({ address, chainId, src, size, setError, ...rest }) {
-  return (
-    <Inline>
-      <Image
-        {...rest}
-        alt={''}
-        src={src}
-        size={size}
-        onError={event => {
-          BAD_IMAGES[chainId] = { ...BAD_IMAGES[chainId], [address]: true }
-          setError(true)
-          event.preventDefault()
-        }}
-      />
-    </Inline>
-  )
-}
-
 export default function TokenLogo({ address, networkInfo, header = false, size = '24px', ...rest }) {
   const [error, setError] = useState(false)
-  const tokensList = useTokensList()?.[networkInfo.chainId]
 
+  const { chainId } = networkInfo
+
+  const tokensList = useTokensList()?.[chainId]
```

Comment:
It would be better to move `[chainId]` into `useTokenlist()` to reduce re-render